src/row_codec.rs

```diff
@@ -1,182 +1,230 @@
+            value: self.value,
+            seq: self.seq,
+            flags: self.flags,
+            create_ts: self.create_ts,
+            expire_ts: self.expire_ts,
+        }
     }
-
-    flags.bits()
 }
 
-/// Decodes a row based on the encoding specified in [`encode_row_v0`]. If the
-/// `data` passed in was not encoded using `encode_row_v0`, or if there are any
-/// unknown row attributes, this method will return an error.
-pub(crate) fn decode_row_v0(
-    first_key: &Bytes,
-    row_features: &[RowFeature],
-    data: &mut Bytes,
-) -> Result<KeyValueDeletable, SlateDBError> {
-    let key_prefix_len = data.get_u16() as usize;
-    let key_suffix_len = data.get_u16() as usize;
-    let key_suffix = data.slice(..key_suffix_len);
-    data.advance(key_suffix_len);
-
-    let meta = decode_meta(row_features, data)?;
-    let value = if !(meta.flags & RowFlags::Tombstone).is_empty() {
-        ValueDeletable::Tombstone
-    } else {
-        let value_len = data.get_u32() as usize;
-        ValueDeletable::Value(data.slice(..value_len))
-    };
-
-    let mut key = BytesMut::with_capacity(key_prefix_len + key_suffix_len);
-    key.extend_from_slice(&first_key[..key_prefix_len]);
-    key.extend_from_slice(&key_suffix);
-
-    Ok(KeyValueDeletable {
-        key: key.into(),
-        value,
-        attributes: RowAttributes {
-            ts: meta.timestamp,
-            expire_ts: meta.expire_ts,
-        },
-    })
-}
+pub(crate) struct SstRowCodecV1 {}
 
-struct RowMetadata {
-    flags: RowFlags,
-    timestamp: Option<i64>,
-    expire_ts: Option<i64>,
-}
+impl SstRowCodecV1 {
+    pub fn new() -> Self {
+        Self {}
+    }
 
-fn decode_meta(row_features: &[RowFeature], data: &mut Bytes) -> Result<RowMetadata, SlateDBError> {
-    let mut meta = RowMetadata {
-        flags: RowFlags::empty(),
-        timestamp: None,
-        expire_ts: None,
-    };
-    for attr in row_features {
-        match attr {
-            RowFeature::Flags => match decode_row_flags(data.get_u8()) {
-                Ok(flags) => {
-                    meta.flags = flags;
-                }
-                Err(e) => return Err(e),
-            },
-            RowFeature::Timestamp => meta.timestamp = Some(data.get_i64()),
-            RowFeature::ExpireAtTs => {
-                let expire_ts = data.get_i64();
-                meta.expire_ts = if expire_ts == NO_EXPIRE_TS {
-                    None
-                } else {
-                    Some(expire_ts)
-                };
+    pub fn encode(&self, output: &mut Vec<u8>, row: &SstRowEntry) {
+        match &row.key {
+            SstRowKey::SuffixOnly { prefix_len, suffix } => {
+                output.put_u16(*prefix_len as u16);
+                output.put_u16(suffix.len() as u16);
+                output.put(suffix.as_ref());
+            }
+            _ => unreachable!("only suffix only key is supported on encode"),
+        }
+
+        let flags = {
```

Comment:
doesn't `row` already have the flags computed? why do we need to recompute them here?

Reply:
yeah it should be computed, i changed it into taking the computed result, PTAL